labgrid/driver/lauterbachdriver.py

```diff
@@ -0,0 +1,143 @@
+import logging
+import os
+import re
+import subprocess
+
+from importlib import import_module
+
+import attr
+
+from ..factory import target_factory
+from ..protocol import DebuggerProtocol
+from ..resource.lauterbach import NetworkLauterbachDebugger, NetworkUSBLauterbachDebugger
+from ..resource.udev import USBLauterbachDebugger
+from ..util.helper import get_uname_machine, processwrapper
+from ..util.managedfile import ManagedFile
+from ..util.proxy import proxymanager
+from ..util import Timeout
+
+from .common import Driver
+
+
+@target_factory.reg_driver
+@attr.s(eq=False)
+class LauterbachDriver(Driver, DebuggerProtocol):
+    """
+    Args:
+        t32_sys (str): base folder of the TRACE32 installation (default ENV['T32SYS'])
+        t32_bin (str): name of the TRACE32 architecture executable `t32m*` (default `t32marm`)
+    """
+    bindings = {
+        "interface": {
+            "USBLauterbachDebugger",
+            "NetworkUSBLauterbachDebugger",
+            "NetworkLauterbachDebugger"
+        },
+    }
+    t32_bin = attr.ib(default="t32marm", validator=attr.validators.instance_of(str))
+    t32_sys = attr.ib(
+        default=None,
+        validator=attr.validators.optional(attr.validators.instance_of(str)))
+
+    def __attrs_post_init__(self):
+        super().__attrs_post_init__()
+        self.logger = logging.getLogger(f"{self}:{self.target}")
+        self._pystart = import_module('lauterbach.trace32.pystart')
+        #import lauterbach.trace32.pystart as pystart
+        self.pathmap = {
+            "aarch64": "bin/linux-aarch64",
+            "amd64":   "bin/pc_linux64",
+            "armhf":   "bin/linux-armhf"
+            }
+        self.t32sys = os.environ.get("T32SYS", "/opt/t32")
```

Comment:
Other drivers use something like:
```python
if self.target.env:
    self.tool = self.target.env.config.get_tool('mybinary')
else:
    self.tool = 'mybinary'
```
Since there is no single binary in this case, maybe use a path from `paths:` via `self.target.env.config.get_path()` from https://labgrid.readthedocs.io/en/latest/configuration.html#environment-configuration instead?

Reply:
The `pystart` library expects only the basename of the executable and handles the rest internally including e.g. host architecture detection. This is put into `t32_bin` now.
The `t32_sys` is used to specify the base installation folder.
I reworked that completely in the latests force push.